interfaces/apparmor/template.go

```diff
@@ -975,6 +975,9 @@ profile snap-update-ns.###SNAP_INSTANCE_NAME### (attach_disconnected) {
   # Allow reading the command line (snap-update-ns uses it in pre-Go bootstrap code).
   @{PROC}/@{pid}/cmdline r,
 
+  # Allow reading of own maps (Go runtime)
+  @{PROC}/@{pid}/maps r,
```

Comment:
I'm actually surprised that we don't have `owner` here and next to the other `@{PROC}/@{pid}/` entries like we have in the template.

Reply:
I think it makes more sense to scope this to `owner` as suggested (and we could do the same for `cmdline` above as well) to make it more explicit that this is about granting permission for snap-update-ns to read its own cmdline/maps etc.

Reply:
I've pushed an update and also added `owner` to fds and cgroup (that one even had a comment about reading own cgroup).